src/plugins/project_file.jl

```diff
@@ -16,14 +16,25 @@ priority(::ProjectFile, ::typeof(hook)) = typemax(Int) - 5
 function hook(p::ProjectFile, t::Template, pkg_dir::AbstractString)
     toml = Dict(
         "name" => basename(pkg_dir),
-        "uuid" => string(uuid4()),
+        "uuid" => string(@mock uuid4()),
         "authors" => t.authors,
         "version" => string(p.version),
         "compat" => Dict("julia" => compat_version(t.julia)),
     )
-    open(io -> TOML.print(io, toml), joinpath(pkg_dir, "Project.toml"), "w")
+    write_project(joinpath(pkg_dir, "Project.toml"), toml)
 end
 
+# Taken from:
+# https://github.com/JuliaLang/Pkg.jl/blob/2341a067973467949653928ab9f49d98cae6057e/src/project.jl#L175-L177
+_project_key_order = ["name", "uuid", "keywords", "license", "desc", "deps", "compat"]
+project_key_order(key::String) =
+    something(findfirst(x -> x == key, _project_key_order), length(_project_key_order) + 1)
```

Comment:
i know this is a direct copy-paste of the linked code, but using a non-const global variable does seem very odd, so i think i prefer we change it for this repo:
```suggestion

function project_key_order(key::String)
    _project_key_order = ["name", "uuid", "keywords", "license", "desc", "deps", "compat"]
    return something(findfirst(x -> x == key, _project_key_order), length(_project_key_order) + 1)
end
```